src/database/model/bookmark/bookmark.py

```diff
@@ -0,0 +1,17 @@
+from sqlmodel import SQLModel, Field
+from typing import Optional
+from datetime import datetime
+from sqlalchemy import Column, String
+
+
+class BookmarkBase(SQLModel):
+    user_identifier: str = Field(
+        foreign_key="user.subject_identifier", nullable=False, ondelete="CASCADE"
+    )
+    resource_identifier: str
+    created_at: datetime = Field(default_factory=datetime.utcnow)
+
+
+class Bookmark(BookmarkBase, table=True):  # type: ignore [call-arg]
+    id: Optional[int] = Field(default=None, primary_key=True)
+    __tablename__ = "bookmark"
```

Comment:
```suggestion
class BookmarkBase(SQLModel):
    user_identifier: str = Field(
        foreign_key="user.subject_identifier", nullable=False, ondelete="CASCADE"
    )
    resource_identifier: str
    created_at: datetime = Field(default_factory=datetime.utcnow)


class Bookmark(BookmarkBase, table=True):  # type: ignore [call-arg]
    id: Optional[int] = Field(default=None, primary_key=True)
    __tablename__ = "bookmark"
```
What is the reason for separation here? I don't think they need to be. And I think it makes a lot of sense to remove the `id` field in favor of making the columns `(user_identifier, resource_identifier)` a primary key together: that way we also automatically make sure a user doesn't add the same bookmark twice.

Reply:
I think you are right. Since I already have separate BookmarkCreate and Read method, so Base method is no more needed.